cfg/cedar/ros-model.clp

```diff
@@ -0,0 +1,24 @@
+
+;---------------------------------------------------------------------------
+;  ros-model.clp - CEDAR example system model for ROS
+;
+;  Created: Fri Dec 06 13:37:54 2013
+;  Copyright  2011-2013  Tim Niemueller [www.niemueller.de]
+;             2011-2013  RWTH Aachen University (KBSG)
+;             2011       SRI International
+;             2011       Carnegie Mellon University
+;  Licensed under GPLv2+ license, cf. LICENSE file of cedar
+;---------------------------------------------------------------------------
+
+(deffacts cedar-model-ros
+  (model-ros-topic (name "/chatter") (type "std_msgs/String"))
+  (model-ros-node  (name "/talkerpub") (published "/chatter" "/rosout") (subscribed)
+		   (services) (flags))
+  (model-ros-node  (name "/talkersub") (published "/rosout") (subscribed "/chatter")
+		   (services) (flags))
+  (model-ros-node  (name "/fawkes") (published "/rosout") (subscribed "/cmd_vel") (services) (flags))
+  (model-ros-topic-connection (topic "/rosout") (from "/fawkes") (to "/rosout"))
+  (model-ros-topic-connection (topic "/chatter") (from "/talkerpub") (to "/talkersub"))
+  ;(model-ros-topic-connection (topic "/rosout") (from "/fawkes") (to "/rostopic_3041_1386770440241"))
+  ;(model-ros-topic-connection (topic "/cmd_vel") (from "/somewhere") (to "/fawkes"))
+)
```

Comment:
Wondering why we need the deffacts in cfg/cedar . Seems a bit unorthodox to me to not have the clips facts defined from the configs so wondering if there is a specific reason for that

Reply:
This is, indeed, not the way we do things today. The reason for this being as it is is that there is another piece of software which generates these models automatically from a running system (for ROS). This stems from even earlier ROS code (this is from 2011 mind you from before there was a CLIPS agent, it technically is an ancestor to that actually) and there was no other flexible configuration format easily queryable from CLIPS. For the Fawkes version, back in the days configuration was done in SQL and there was no suitable (and readable) format to keep the model. Hence, the best and most flexible way to represent it was to generate a CLIPS file that contains the model. So this really is system-specific configuration. It would have to change for each specific system you run this on and hence resides in `cfg/cedar`.

If you were to write or update this today, it would probably be based on a YAML configuration.

Reply:
I see. What is this piece of software though, its not a plugin is it?.is it still relevant?
if not, then why not update this to load from YAML to be more consistent with what we do today.

Reply:
As I said before, this is more of a suggestion. I still approve the PR since its really up to you if you want to keep that

Reply:
It is a plugin to run the CLIPS context.

I'm cleaning out my "code basement". The code was relevant at a time, and could be again if someone picks it up. In a sense, it can provide system-level monitoring that could be used as input in the CX, e.g., to automatically call for help if crucial components fail. It was also used to build fault trees or detect stuck camera drivers.

Reply:
That sounds very interesting actually. Can you perhaps point me to that plugin.

Reply:
That sounds very interesting actually. Can you perhaps point me to that plugin.

Reply:
Ok, there was a misunderstanding. I thought you asked why this is a plugin, it's one to embed the CLIPS context. After reading this again, it seems you are asking for the code that generates the models? Well, there I yet have to check whether I can make this available (due to contract obligations from back then).